include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1126,7 +1151,9 @@ __parallel_find_or(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPoli
         oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<__find_or_kernel, _CustomName, _Brick,
                                                                                _BrickTag, _Ranges...>;
 
-    constexpr bool __or_tag_check = ::std::is_same_v<_BrickTag, __parallel_or_tag>;
+    assert("This device does not support 64-bit atomics" &&
+           (sizeof(_AtomicType) < 64 || __exec.queue().get_device().has(sycl::aspect::atomic64)));
```

Comment:
`sizeof` uses bytes so this should be `sizeof(_AtomicType) < 8`. I think even better would be `<= 4` since only 32-bit data types are then allowed.

Reply:
Good catch, @julianmi 
But this mean we have the same error here:
```C++
    assert("This device does not support 64-bit atomics" &&
           (sizeof(_Type) < 64 || __queue.get_device().has(sycl::aspect::atomic64)));
```
I mean https://github.com/oneapi-src/oneDPL/blob/e1fee6404e4ebda93ade0d3e913e1eb9e9b8595b/include/oneapi/dpl/experimental/kt/single_pass_scan.h#L328
It was introduced in https://github.com/oneapi-src/oneDPL/commit/e625bf013767577392c3b3e28fa1d8e46112f3fb

Reply:
Fixed.